ngraph/core/reference/include/ngraph/runtime/reference/random_uniform.hpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ctime>
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            std::pair<uint32_t, uint32_t> split_high_low(uint64_t value)
+            {
+                uint32_t low = static_cast<uint32_t>(value);
+                uint32_t high = static_cast<uint32_t>(value >> 32);
+                return {low, high};
+            }
+
+            uint64_t unite_high_low(uint32_t high, uint32_t low)
+            {
+                return (static_cast<uint64_t>(high) << 32) + low;
+            }
+
+            void calculate_round(const uint64_t& key, uint64_t& counter, uint64_t& n)
+            {
+                auto counter_lr = split_high_low(counter);
+                auto key_lr = split_high_low(key);
+                auto n_lr = split_high_low(n);
+                auto prod0 = split_high_low(static_cast<uint64_t>(0xD2511F53) * n_lr.first);
+                auto prod1 = split_high_low(static_cast<uint64_t>(0xCD9E8D57) * counter_lr.first);
+
+                n_lr.first = prod1.second ^ n_lr.second ^ key_lr.first;
+                n_lr.second = prod1.first;
+                counter_lr.first = prod0.second ^ counter_lr.second ^ key_lr.second;
+                counter_lr.second = prod0.first;
+
+                counter = unite_high_low(counter_lr.second, counter_lr.first);
+                n = unite_high_low(n_lr.second, n_lr.first);
+            }
+
+            void raise_key(uint64_t& key)
+            {
+                auto key_lr = split_high_low(key);
+                key_lr.first += 0x9E3779B9;
+                key_lr.second += 0xBB67AE85;
+                key = unite_high_low(key_lr.second, key_lr.first);
+            }
+
+            float uint32_to_float(uint32_t x)
+            {
+                uint32_t x_uint32 = (static_cast<uint32_t>(127) << 23) | (x & 0x7fffffu);
+
+                float x_float;
+                memcpy(&x_float, &x_uint32, sizeof(x_uint32));
+                return x_float - 1.0f;
+            }
+
+            float16 uint32_to_float16(uint32_t x)
+            {
+                uint16_t x_uint16 = static_cast<uint16_t>(x);
+                x_uint16 = (static_cast<uint16_t>(15) << 10) | (x_uint16 & 0x3ffu);
+
+                float16 x_float16;
+                memcpy(&x_float16, &x_uint16, sizeof(x_uint16));
+                return x_float16 - static_cast<float16>(1);
+            }
+
+            double uint32_to_double(uint32_t x1, uint32_t x2)
+            {
+                uint64_t mantissa =
```

Comment:
`mantissa` ->`significant` ? 😄

Reply:
Applied this comment in separate PR: https://github.com/openvinotoolkit/openvino/pull/7012